sources/packages/backend/libs/integrations/src/institution-integration/ece-integration/ece-response.processing.service.ts

```diff
@@ -108,40 +112,32 @@ export class ECEResponseProcessingService {
         this.logger.log(warningMessage);
         return processSummary;
       }
+      // Set the total disbursements count.
+      disbursementProcessingDetails.totalDisbursements =
+        eceFileDetailRecords.length;
       // Sanitize all the ece response detail records.
-      this.sanitizeDisbursements(eceFileDetailRecords, processSummary);
+      this.sanitizeDisbursements(
+        eceFileDetailRecords,
+        processSummary,
+        disbursementProcessingDetails,
+      );
       // Transform ece response detail records to disbursements which could be individually processed.
       const disbursementsToProcess =
         this.transformDetailRecordsToDisbursements(eceFileDetailRecords);
       const auditUser = await this.systemUsersService.systemUser();
-      const {
-        totalDisbursements,
-        disbursementsSuccessfullyProcessed,
-        disbursementsSkipped,
-        duplicateDisbursements,
-        disbursementsFailedToProcess,
-      } = await this.validateAndUpdateEnrolmentStatus(
+      await this.validateAndUpdateEnrolmentStatus(
         disbursementsToProcess,
         auditUser.id,
         processSummary,
+        disbursementProcessingDetails,
       );
-      processSummary.summary.push(
-        `Total disbursements found: ${totalDisbursements}`,
-      );
-      processSummary.summary.push(
-        `Disbursements successfully updated: ${disbursementsSuccessfullyProcessed}`,
-      );
-      processSummary.summary.push(
-        `Disbursements skipped to be processed: ${disbursementsSkipped}`,
-      );
-      processSummary.summary.push(
-        `Disbursements considered duplicate and skipped: ${duplicateDisbursements}`,
-      );
-      processSummary.summary.push(
-        `Disbursements failed to process: ${disbursementsFailedToProcess}`,
-      );
+
       this.logger.log(`Completed processing the file ${remoteFilePath}.`);
     } catch (error: unknown) {
+      // If error is thrown while downloading the file processing errors must be incremented.
+      if (!disbursementProcessingDetails.fileParsingErrors) {
```

Comment:
Sorry, I did not follow the `if` condition here. Can you please elaborate on that?

Reply:
Updated the comments to be more precise, 

![image](https://github.com/bcgov/SIMS/assets/54600590/5dd7ba50-306b-49bf-97f1-6f842c39093a)